app/src/main/java/org/simple/clinic/settings/changelanguage/ChangeLanguageLogic.kt

```diff
@@ -0,0 +1,18 @@
+package org.simple.clinic.settings.changelanguage
+
+import com.spotify.mobius.First
+import com.spotify.mobius.First.first
+
+object ChangeLanguageLogic {
+
+  fun init(model: ChangeLanguageModel): First<ChangeLanguageModel, ChangeLanguageEffect> {
+    val effects = when {
+      model.currentLanguage == null && model.supportedLanguages.isEmpty() -> setOf(LoadCurrentSelectedLanguageEffect, LoadSupportedLanguagesEffect)
+      model.currentLanguage == null -> setOf(LoadCurrentSelectedLanguageEffect)
+      model.supportedLanguages.isEmpty() -> setOf(LoadSupportedLanguagesEffect)
+      else -> emptySet()
+    }
+
+    return first(model, effects)
```

Comment:
Can we make this `if (effects.isNotEmpty()) first(model, effects) else first(model)`?